main.tf

```diff
@@ -29,6 +29,19 @@ resource "aws_grafana_workspace" "this" {
   tags = var.tags
 }
 
+################################################################################
+# Workspace API Key
+################################################################################
+
+resource "aws_grafana_workspace_api_key" "this" {
+  for_each = { for k, v in var.workspace_api_keys : k => v if var.create }
```

Comment:
Maybe it should check var.create_workspace instead?

Reply:
[this](https://github.com/terraform-aws-modules/terraform-aws-managed-service-grafana/pull/9/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR42) handles that but slightly differently. 

tl;dr - users can create a workspace and all the things here, or they can create all the things here (minus a workspace) on an existing workspace